modules/FindImGui.cmake

```diff
@@ -0,0 +1,101 @@
+#   Copyright © 2018 Jonathan Hale <squareys@googlemail.com>
+#
+#   Permission is hereby granted, free of charge, to any person obtaining a
+#   copy of this software and associated documentation files (the "Software"),
+#   to deal in the Software without restriction, including without limitation
+#   the rights to use, copy, modify, merge, publish, distribute, sublicense,
+#   and/or sell copies of the Software, and to permit persons to whom the
+#   Software is furnished to do so, subject to the following conditions:
+#
+#   The above copyright notice and this permission notice shall be included
+#   in all copies or substantial portions of the Software.
+#
+#   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+#   THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+#   FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+#   DEALINGS IN THE SOFTWARE.
+#
+
+list(FIND ImGui_FIND_COMPONENTS "Sources" _INDEX)
+if(_INDEX GREATER -1)
+    set(_FIND_SOURCES TRUE)
+endif()
+
+if(NOT TARGET ImGui::ImGui)
+    find_package(imgui CONFIG)
+    if(imgui_FOUND)
+        add_library(ImGui::ImGui INTERFACE IMPORTED)
+        set_property(TARGET ImGui::ImGui APPEND PROPERTY
+            INTERFACE_LINK_LIBRARIES imgui::imgui)
+
+        get_target_property(ImGui_INCLUDE_DIR imgui::imgui
+            INTERFACE_INCLUDE_DIRECTORIES)
+
+        add_library(ImGui::Sources INTERFACE IMPORTED)
+        set_property(TARGET ImGui::Sources APPEND PROPERTY
+            INTERFACE_LINK_LIBRARIES ImGui::ImGui)
+    else()
+        find_path(ImGui_INCLUDE_DIR NAMES imgui.h HINTS ${IMGUI_DIR})
+
+        add_library(ImGui::ImGui INTERFACE IMPORTED)
+        set_property(TARGET ImGui::ImGui APPEND PROPERTY
+            INTERFACE_INCLUDE_DIRECTORIES "${ImGui_INCLUDE_DIR}")
+    endif()
+endif()
+
+# Find components
+foreach(_component IN LISTS ImGui_FIND_COMPONENTS)
+
+    if(_component STREQUAL "Sources")
+        if(NOT TARGET ImGui::Sources)
+            find_file(ImGui_SOURCE NAMES imgui.cpp HINTS ${IMGUI_DIR})
+            set(ImGui_SOURCES ${ImGui_SOURCE})
+
+            foreach(_file widgets draw demo)
+                find_file(ImGui_${_file}_SOURCE NAMES imgui_${_file}.cpp
+                    HINTS ${IMGUI_DIR})
+                list(APPEND ImGui_SOURCES ${ImGui_${_file}_SOURCE})
+            endforeach()
+
+            add_library(ImGui::Sources INTERFACE IMPORTED)
+            set_property(TARGET ImGui::Sources APPEND PROPERTY
+                INTERFACE_SOURCES "${ImGui_SOURCES}")
+            set_property(TARGET ImGui::Sources APPEND PROPERTY
+                INTERFACE_LINK_LIBRARIES ImGui::ImGui)
+
+            set(ImGui_Sources_FOUND ${ImGui_SOURCES})
```

Comment:
@mosra How would I go about this? I though about a couple of if/else to set to TRUE or FALSE, but that sounds cumbersome. Maybe check the list length or something?

Reply:
Set it to `ON` by default, then in the foreach above set it to `OFF` if `NOT ImGui_${_file}_SOURCE`.

Besides that, `if(NOT TARGET)` should wrap only the target addition, not the whole thing.